native/core/src/parquet/read/levels.rs

```diff
@@ -121,8 +121,8 @@ impl LevelDecoder {
             match self.mode {
                 Mode::RLE => {
                     if self.current_value as i16 == max_def_level {
-                        value_decoder.read_batch(vector, n);
                         bit::set_bits(vector.validity_buffer.as_slice_mut(), vector.num_values, n);
+                        value_decoder.read_batch(vector, n);
```

Comment:
I don't see how this (or the similar change in line 136 below) make a difference

Reply:
`value_decoder.read_batch` may unset `validity_buffer` for decimal values if the value is larger than precision limit, so that those values be understood as null.

Reply:
Got it. Thanks